web/src/main/java/org/springframework/security/web/util/matcher/RequestMatchers.java

```diff
@@ -60,7 +83,236 @@ public static RequestMatcher not(RequestMatcher matcher) {
 		return (request) -> !matcher.matches(request);
 	}
 
+	/**
+	 * Create {@link RequestMatcher}s whose URIs do not have a servlet path prefix
+	 * <p>
+	 * When there is no context path, then these URIs are effectively absolute.
+	 * @return a {@link Builder} that treats URIs as relative to the context path, if any
+	 * @since 6.5
+	 */
+	public static Builder request() {
+		return new Builder();
+	}
+
+	/**
+	 * Create {@link RequestMatcher}s whose URIs are relative to the given
+	 * {@code servletPath}.
+	 *
+	 * <p>
+	 * The {@code servletPath} must correlate to a configured servlet in your application.
+	 * The path must be of the format {@code /path}.
+	 * @return a {@link Builder} that treats URIs as relative to the given
+	 * {@code servletPath}
+	 * @since 6.5
+	 */
+	public static Builder servlet(String servletPath) {
+		Assert.notNull(servletPath, "servletPath cannot be null");
+		Assert.isTrue(servletPath.startsWith("/"), "servletPath must start with '/'");
+		Assert.isTrue(!servletPath.endsWith("/"), "servletPath must not end with a slash");
+		Assert.isTrue(!servletPath.contains("*"), "servletPath must not contain a star");
```

Comment:
I know what I should do, but to me it is not very clear to users what to do if a Servlet is mapped to `*.do`.

Reply:
What do they do if it has a mapping of `/foo` and `*.do`?

Reply:
What does a user do if the servlet maps to `/foo/*`?

Reply:
They will need to use something else. This is for the purpose of matching `HttpServletRequest#getServletPath`. I think a method name change will be helpful.

Note that I haven't yet been able to get Framework's `PathPattern` to work with extensions, so that may be a follow-up item. As it stands, this specific method doesn't comment on that.